src/session/content/chat_action_bar.rs

```diff
@@ -575,6 +577,84 @@ impl ChatActionBar {
 
         self.notify("reply-to-message-id");
     }
+
+    pub(crate) fn update_stack_page(&self) {
+        if let Some(chat) = self.chat() {
+            match chat.type_() {
+                ChatType::Private(user) => {
+                    let is_deleted = matches!(user.type_().0, UserType::Deleted);
+                    let is_blocked = chat.is_blocked();
+                    if is_deleted {
+                        // TODO: Add delete chat button
+                        self.imp().action_bar_stack.set_visible_child_name("entry");
```

Comment:
To save quite a bit of code, can you put `let imp = self.imp();` at the start of the function and then use that instead of using `self.imp()` every time? That's a pattern we use everywhere.